runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java

```diff
@@ -199,30 +223,40 @@ String getEstimatedRunTimeRemaining(boolean inSeconds)
 		}
 	}
 
-	private boolean isLocalPlayerWearingFullGraceful()
+	private int getGracefulRecoveryBoost()
 	{
 		final ItemContainer equipment = client.getItemContainer(InventoryID.EQUIPMENT);
 
 		if (equipment == null)
 		{
-			return false;
+			return 0;
 		}
 
 		final Item[] items = equipment.getItems();
 
-		// Check that the local player is wearing enough items to be using full Graceful
-		// (the Graceful boots will have the highest slot index in the worn set).
-		if (items == null || items.length <= EquipmentInventorySlot.BOOTS.getSlotIdx())
+		if (items == null)
 		{
-			return false;
+			return 0;
 		}
 
-		return (ALL_GRACEFUL_HOODS.contains(items[EquipmentInventorySlot.HEAD.getSlotIdx()].getId()) &&
-			ALL_GRACEFUL_TOPS.contains(items[EquipmentInventorySlot.BODY.getSlotIdx()].getId()) &&
-			ALL_GRACEFUL_LEGS.contains(items[EquipmentInventorySlot.LEGS.getSlotIdx()].getId()) &&
-			ALL_GRACEFUL_GLOVES.contains(items[EquipmentInventorySlot.GLOVES.getSlotIdx()].getId()) &&
-			ALL_GRACEFUL_BOOTS.contains(items[EquipmentInventorySlot.BOOTS.getSlotIdx()].getId()) &&
-			ALL_GRACEFUL_CAPES.contains(items[EquipmentInventorySlot.CAPE.getSlotIdx()].getId()));
+		int boost = 0;
+
+		for (final GracefulEquipmentSlot slot : GracefulEquipmentSlot.values())
+		{
+			final int wornItemId = items[slot.getIndex()].getId();
```

Comment:
This can cause `ArrayIndexOutOfBoundsException` as `equipment.getItems()` is not guaranteed to have a length out to the highest equipment slot type's ID. Moving the `items.length > slot.getIndex()` check which is currently part of the if statement below this to the start of the block should correct that.

```
if (items.length <= slot.getIndex())
{
	continue;
}

// rest of method
```